torch/_inductor/ir.py

```diff
@@ -6213,3 +6256,106 @@ def maybe_free_unbacked_symbols(s):
         return free_unbacked_symbols(s)
     else:
         return set()
+
+
+class AllToAllSingle(OutOfPlaceCollectiveKernel):
+    """
+    NOTE: Difference of AllToAllSingle compared to other OutOfPlaceCollectiveKernel:
+
+    1. `create_output_buffers` is not called, because we rely on the underlying
+    `fun_col_impl._all_to_all_single` kernel to do the output buffer allocation.
+    2. `fun_col_impl._register_tensor_work` is not emitted, because the underlying
+    `fun_col_impl._all_to_all_single` kernel already calls it.
+    3. `c10d._find_or_create_pg_by_ranks_and_tag` is not emitted, because the underlying
+    `fun_col_impl._all_to_all_single` kernel already calls it.
+    """
+
+    def __init__(
+        self,
+        layout,
+        inputs,
+        constant_args,
+        output_split_sizes,
+        input_split_sizes,
+    ):
+        super().__init__(layout, inputs, [], constant_args)
+        self.output_split_sizes = output_split_sizes
+        self.input_split_sizes = input_split_sizes
+
+    def get_unbacked_symbol_uses(self):
+        r = set()
+        if self.output_split_sizes is not None:
+            r |= free_unbacked_symbols(self.output_split_sizes)
+        if self.input_split_sizes is not None:
+            r |= free_unbacked_symbols(self.input_split_sizes)
+        return r
+
+    def should_emit_register_tensor_work(self):
+        return False
+
+    def should_emit_find_or_create_pg(self):
+        return False
```

Comment:
@yf225 can you explain why you needed to disable register tensor work / find-create pg? This seems surprising to me